### PR TITLE
Add more drivebase capabilites, support for Crane in Drivebase

### PR DESCRIPTION
**Pull request by jbko6**

### CODE REVIEW

**Comment by jamesdooley4** [comment]
@jbko6  - did you intend to bring back the local copy of YAGSL? Fine if you did, I just wanted to double-check

File: src/main/java/frc/team2412/robot/Robot.java [hunk 1]
```diff
@@ -129,4 +133,10 @@ public RobotType getRobotType() {
 	public boolean isCompetition() {
 		return getRobotType() == RobotType.COMPETITION;
 	}
+
+	public void registerCommands() {
+		// parameters: Str command name, Command command
+
+		// NamedCommands.registerCommand(null, null);
```
Comment (jamesdooley4) [inline src/main/java/frc/team2412/robot/Robot.java:140]: Please add a comment noting that these are commands intended for use in Autonomous, so not every single command gets listed here.

File: src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java [hunk 2]
```diff
@@ -8,46 +8,78 @@
...
 	private static final double MAX_AUTO_SPEED =
 			500.0; // this seems to only affect rotation for some reason
 
 	private final SwerveDrive swerveDrive;
+	private final ShuffleboardTab drivebaseTab = Shuffleboard.getTab("Drivebase");
+
+	private boolean xWheelsEnabled = true;
+
+	// shuffleboard variables
+	private GenericEntry headingCorrectionEntry;
```
Comment (jamesdooley4) [inline src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java:63]: Should be `final` so the compiler checks it was initialized during class construction

**Review by jamesdooley4 — APPROVED** [review]

File: networktables.json.bck
Comment (TAKBS2412) [inline networktables.json.bck:1]: I'm curious, what's this file for?

File: src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java [hunk 3]
```diff
@@ -57,15 +89,12 @@ public DrivebaseSubsystem() {
 			throw new RuntimeException();
 		}
 
-		// set drive motors to brake
 		swerveDrive.setMotorIdleMode(true);
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java:92]: (no action required) I do wish swervelib renamed this method or added an enum to make it clear that `true` means brake mode...

**Review by TAKBS2412 — APPROVED** [review]

**Comment by jbko6** [comment]
> @jbko6 - did you intend to bring back the local copy of YAGSL? Fine if you did, I just wanted to double-check

Yes I needed it back to gain a little more control over heading correction deadbands.

**Comment by jbko6** [comment]
I'm going to split this PR into one that adds yagsl support for Crane and one that adds improved heading correction & additional drivebase features.

**Comment by TAKBS2412** [comment]
> > @jbko6 - did you intend to bring back the local copy of YAGSL? Fine if you did, I just wanted to double-check
> 
> Yes I needed it back to gain a little more control over heading correction deadbands.

Additionally, the yagsl vendordep is hosted on github.io, which is blocked on school computers.

**Pull request closed without merging**